src/main/java/frc/robot/OI.java

```diff
@@ -18,225 +19,191 @@ public class OI {
+      this(axis.value, action, axis.name(), controller, ControlType.AXIS);
+      this.curve = curve;
     }
 
-    public static JoystickButton getOrientationButton() {
-      return orientationButton.buildButton("Toggle swerve orientation");
+    Control(XboxController.Axis axis, String action, XboxController controller) {
+      this(axis, action, controller, new ControlCurve(1, 0, 0, 0));
     }
 
-    public static Trigger getIntakeButton() {
-      return intakeButton.buildTrigger("Intake");
+    Control(XboxController.Button button, String action, XboxController controller) {
+      this(button.value, action, button.name(), controller, ControlType.BUTTON);
     }
 
-    public static JoystickButton getOuttakeButton() {
-      return outtakeButton.buildButton("Outtake");
+    Control(double povAngle, String action, XboxController controller) {
+      this((int) povAngle, action, "POV " + povAngle, controller, ControlType.POVBUTTON);
     }
 
-    public static JoystickButton getBrakeButton() {
-      return brakeButton.buildButton("Brake");
+    Control(XboxController.Axis axis, String action, XboxController controller, double detent) {
+      this(axis.value, action, axis.name(), controller, ControlType.TRIGGER);
+      this.detent = detent;
     }
 
-    public static JoystickButton getResetRotationButton() {
-      return resetRotationButton.buildButton("Reset Rotation");
+    private int getId() {
+      return id;
     }
 
-    public static void setRumble(double rumbleIntensity) {
-      rumbleController.setRumble(RumbleType.kBothRumble, rumbleIntensity);
+    private String getAction() {
+      return action;
     }
-  }
 
-  public static final class Operator {
-    private static enum Button {
-      A(1),
-      B(2),
-      X(3),
-      Y(4),
-      LB(5), // Left Bumper
-      RB(6), // Right Bumper
-      LT(2, 0.5), // Left Trigger
-      RT(3, 0.5), // Right Trigger
-      Back(7),
-      Start(8),
-      LJ(9), // Left Joystick Button
-      RJ(10), // Right Joystick Button
-      POVUP(0),
-      POVDOWN(180),
-      POVLEFT(270),
-      POVRIGHT(90);
-
-      private final int buttonID;
-      private String buttonAction;
-      private double triggerDetent; // Percentage where axis is triggered as a button
-
-      Button(int ID) {
-        this.buttonID = ID;
-        this.buttonAction = "";
-      }
+    private String getName() {
+      return name;
+    }
 
-      Button(int ID, double axisDetent) {
-        this(ID);
-        this.triggerDetent = axisDetent;
-      }
+    private XboxController getController() {
+      return controller;
+    }
 
-      private int getButtonID() {
-        return this.buttonID;
-      }
+    private ControlCurve getCurve() {
+      return curve;
+    }
 
-      private double getTriggerDetent() {
-        return this.triggerDetent;
-      }
+    private double getDetent() {
+      return detent;
+    }
```

Comment:
Spelling? Or is this a word I don't know?

Reply:
Maybe threshold is a better word, I'm just used to it being called a detent in triggers

Reply:
agreed.  Threshold is better.